### PR TITLE
Fix handling of Package

### LINKED ISSUE #203: Support Package
- Check https://github.com/objectionary/eo/issues/2940
- Evaluate the “Package" function to same object without it (and dataize it recursively)?

### LINKED ISSUE #222: Fix mathjax expressions on the site
See the [docs](https://rust-lang.github.io/mdBook/format/mathjax.html).

### PR DESCRIPTION
**Pull request by fizruk**
In addition to fixing how `λ ⤍ Package` is evaluated, this PR refactors the "chain" versions of dataization functions to use a new dedicated monad to implicitly propagate the context and separate writing logs from returning results. The non-chain variants have been simplified to use the chain variants, simply ignoring the logs.

Closes #203, closes #222

---

## PR-Codex overview
The focus of this PR is to enhance dataization in the `eo-phi-normalizer` by adding support for new object types and improving parsing functions.

### Detailed summary
- Added support for new object types like `bool` and `bytes` in dataization rules
- Updated object parsing functions to handle new object types
- Improved error handling in parsing functions for `Object` and `Program` types
- Enhanced dataization process to handle `Package` lambdas and siblings efficiently

> The following files were skipped due to too many changes: `eo-phi-normalizer/src/Language/EO/Phi/Rules/Common.hs`, `eo-phi-normalizer/src/Language/EO/Phi/Dataize.hs`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: eo-phi-normalizer/app/Main.hs [hunk 1]
```diff
@@ -353,7 +353,8 @@ main = do
       unless (single || json) $ logStrLn ruleSet.title
       let Program bindings = program'
           uniqueResults
-            | chain = applyRulesChainWith limits ctx (Formation bindings)
+            -- Something here seems incorrect
+            | chain = pure $ fst $ applyRulesChainWith' limits ctx (Formation bindings)
```
Comment (aabounegm) [inline eo-phi-normalizer/app/Main.hs:357]: Please check if this makes sense

File: eo-phi-normalizer/src/Language/EO/Phi/Rules/Common.hs [hunk 2]
```diff
@@ -219,21 +233,74 @@ equalBinding (AlphaBinding attr1 obj1) (AlphaBinding attr2 obj2) = attr1 == attr
...
+        else do
+          let appliedRules = applyOneRule ctx obj
+          forM appliedRules $ \(ruleName, obj') -> do
+            logStep ruleName obj'
+            chain <-
+              if objectSize obj' < maxTermSize
+                then applyRulesChainWith limits{maxDepth = maxDepth - 1} obj'
+                else return [[obj']]
+            -- TODO: verify if this is actually correct. I suspect that it will merge logs of multiple paths together into one
+            return (concat chain)
```
Comment (aabounegm) [inline eo-phi-normalizer/src/Language/EO/Phi/Rules/Common.hs:303]: I'm doubting the correctness of this line as well

**Review by aabounegm — APPROVED** [review]
